superset/utils/pandas_postprocessing.py

```diff
@@ -264,6 +264,17 @@ def pivot(  # pylint: disable=too-many-arguments
     #  Remove once/if support is added.
     aggfunc = {na.column: na.aggfunc for na in aggregate_funcs.values()}
 
+    # When dropna = False, the pivot_table function will calculate cartesian-product
+    # for MultiIndex.
+    # https://github.com/apache/superset/issues/15956
+    # https://github.com/pandas-dev/pandas/issues/18030
+    series_set = set()
+    lst_to_str: Callable[[List[Any]], str] = lambda lst: "_".join(str(_) for _ in lst)
```

Comment:
minor nit - I think you could just replace calls of this this with `str()`.

Reply:
The value of columns might integer or float, so needs to be converted.

Reply:
Yeah, but the join is not necessary - you can just call `str()` on a list to get a (deterministic) string representation of the list.

```
In [1]: str([0,"x"])
Out[1]: "[0, 'x']"
```

`tuple()` might be better (I assume the intention is to make the list hashable)

Reply:
nice tips! changes done.